File: src/main/java/no/novari/flyt/configuration/validation/constraints/AtLeastOneChild.java
package no.novari.flyt.configuration.validation.constraints;

import jakarta.validation.Constraint;
import jakarta.validation.Payload;

import java.lang.annotation.Retention;

import static java.lang.annotation.RetentionPolicy.RUNTIME;

@Retention(RUNTIME)
@Constraint(validatedBy = AtLeastOneChildValidator.class)
public @interface AtLeastOneChild {

    String message() default "has no children";

    Class<?>[] groups() default {};

    Class<? extends Payload>[] payload() default {};

}

File: src/main/kotlin/no/novari/flyt/configuration/validation/constraints/AtLeastOneChild.kt
package no.novari.flyt.configuration.validation.constraints

import jakarta.validation.Constraint
import jakarta.validation.Payload
import kotlin.reflect.KClass

@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS)
@Constraint(validatedBy = [AtLeastOneChildValidator::class])
annotation class AtLeastOneChild(
    val message: String = "has no children",
    val groups: Array<KClass<*>> = [],
    val payload: Array<KClass<out Payload>> = [],
)
